Sources/PostgreSQL/Database/PostgreSQLDatabaseConfig.swift

```diff
@@ -32,47 +18,43 @@ public struct PostgreSQLDatabaseConfig {
     /// Optional password to use for authentication.
     public let password: String?
     
+    /// Which server to connect to.
+    public let serverAddress: PostgreSQLConnection.ServerAddress
+    
     /// Configures how data is transported to the server. Use this to enable SSL.
     /// See `PostgreSQLTransportConfig` for more info
-    public let transportConfig: PostgreSQLTransportConfig
+    public let transportConfig: PostgreSQLConnection.TransportConfig
+    
+    public init(hostname: String, port: Int = 5432, username: String, database: String? = nil, password: String? = nil, transport: PostgreSQLConnection.TransportConfig = .cleartext) {
+        self.init(serverAddress: .tcp(hostname: hostname, port: port), username: username, database: database, password: password, transport: transport)
+    }
     
-    /// Creates a new `PostgreSQLDatabaseConfig`.
-	public init(serverAddress: ServerAddress, username: String, database: String? = nil, password: String? = nil, transport: PostgreSQLTransportConfig = .cleartext) {
-		self.serverAddress = serverAddress
+	public init(serverAddress: PostgreSQLConnection.ServerAddress, username: String, database: String? = nil, password: String? = nil, transport: PostgreSQLConnection.TransportConfig = .cleartext) {
 		self.username = username
 		self.database = database
 		self.password = password
+        self.serverAddress = serverAddress
 		self.transportConfig = transport
 	}
-	
-	public init(hostname: String, port: Int = 5432, username: String, database: String? = nil, password: String? = nil, transport: PostgreSQLTransportConfig = .cleartext) {
-		self.init(serverAddress: .tcp(hostname: hostname, port: port), username: username, database: database, password: password, transport: transport)
-	}
 
-    /// Creates a `PostgreSQLDatabaseConfig` frome a connection string.
-    public init(url urlString: String, transport: PostgreSQLTransportConfig = .cleartext) throws {
+    public init(url urlString: String, transport: PostgreSQLConnection.TransportConfig = .cleartext) throws {
         guard let url = URL(string: urlString),
             let hostname = url.host,
             let port = url.port,
             let username = url.user,
+            let database = url.databaseName,
             url.path.count > 0
```

Comment:
This check (`url.path.count > 0`) is probably not necessary since the previous line will already guarantee that database name was supplied as a part of the connection string?